app/javascript/src/lib/OWLanguageLinter.ts

```diff
@@ -242,40 +244,106 @@ function checkMixins(content: string): void {
-      if (phrase?.text.includes("include")) return
-      if (phrase?.text && !acceptedPhrases.includes(phrase.text)) throw new Error(`Using @translate inside of "${phrase.text}" has no effect.`)
-    } catch (error: any) {
+    // Find translations that are not in client side actions
+    const phrase = getPhraseFromPosition(line, parenthesisBeforeAtIndex - 1)
+    const acceptedPhrases = ["Create HUD Text", "Create In-World Text", "Create Progress Bar HUD Text", "Create Progress Bar In-World Text", "Set Objective Description", "Big Message", "Small Message"]
+    if (phrase?.text.includes("include")) continue
+    if (phrase?.text && !acceptedPhrases.includes(phrase.text)) {
       diagnostics.push({
         from: match.index,
-        to: match.index + match[0].length,
+        to: match.index + match[1].length,
         severity: "warning",
-        message: error.message
+        message: `Using @translate inside of "${phrase.text}" has no effect.`
       })
+      continue
+    }
+
+    // Check translate parameters
+    const translateStartParenthesisIndex = match.index + match[0].length - 1
+    const translateEndParenthesisIndex = getClosingBracket(content, "(", ")", translateStartParenthesisIndex - 1)
+    if (translateEndParenthesisIndex === -1) continue
+
+    const translateArguments = splitArgumentsString(content.substring(translateStartParenthesisIndex + 1, translateEndParenthesisIndex - 1))
+    if (translateArguments.length === 0) continue
+
+    if (!translateArguments[0].startsWith("\"") || !translateArguments[0].endsWith("\"")) {
+      diagnostics.push({
+        from: match.index + match[0].length,
+        to: match.index + match[0].length + translateArguments[0].length,
+        severity: "error",
+        message: "Key argument of @translate must be a string."
+      })
+      continue
+    }
+
+    // Provide option to create translation key via linter
+    const translateKey = translateArguments[0].substring(1, translateArguments[0].length - 1)
+    const hasTranslateKey = translateKey in get(translationKeys)
+    if (!hasTranslateKey) {
+      diagnostics.push({
+        from: match.index,
+        to: match.index + match[1].length,
+        severity: "warning",
+        message: "Unknown translate key.",
+        actions: [{
+          name: "Create translation key",
+          apply() {
+            const newTranslation: TranslationKey = {}
+            const currentDefaultLanguage = get(defaultLanguage)
+
+            if (currentDefaultLanguage) {
+              newTranslation[currentDefaultLanguage] = translateKey
+            }
+
+            translationKeys.update((keys) => ({
+              ... keys,
+              [translateKey]: newTranslation
+            }))
+
+            modal.show(Modal.TranslationKeys, {
+              parameters: {
```

Comment:
I think it might more sense to call this `props`, as that is what the Svelte component parameters are called

Reply:
_An abbreviation?? In this code base?!_

---

Ah, I didn't remember how Svelte called them, and I was too lazy to look it up.
Sure, I'll change it.